autogen.pl

```diff
@@ -1109,6 +1119,23 @@ sub patch_autotools_output {
         $c =~ s/$search_string/$replace_string/g;
     }
 
+    $search_string = 'case \$cc_basename in
+     ifort\*\|nagfor\*\) _lt_dar_can_shared=yes ;;
+     \*\) _lt_dar_can_shared=\$GCC ;;
+  esac'; 
+    $replace_string = "case \$cc_basename in
+     flang*|ifort*|nagfor*) _lt_dar_can_shared=yes ;;
+     *) _lt_dar_can_shared=\$GCC ;;
+  esac"; 
+    push(@verbose_out, $indent_str . "Patching configure for flang compiler on Darwin (FC)\n");
+    $c =~ s/$search_string/$replace_string/g;
+
+    $search_string = 'archive_cmds_FC="\\\\\\$CC -dynamiclib';
+    $replace_string = 'archive_cmds_FC="\$CC --shared';
+    push(@verbose_out, $indent_str . "Patching configure for flang compiler on Darwin (FC)\n");
+    $c =~ s/$search_string/$replace_string/g;
+    $c =~ s/(archive_cmds_FC.*)-install_name \\\$rpath/$1-Wl,-install_name,\\\`echo \\\$rpath | sed \'s%^[ ]*%%\'\\\`/g;
```

Comment:
I confirm that this PR works for me, but can you explain what this pair of search/replaces do?

Reply:
`-dynamiclib` is not supported by `flang` in order to create a shared library, so use `--shared` instead.

`-install_name <lib>` is not supported by `flang`, so replace it with `-Wl,-install_name,lib`

for some reason `$rpath` starts with a (some?) blank space, so trim is otherwise `-Wl,-install_name, lib` fails.